File: src/main/java/sc02/Q10.java
package sc02;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Q10 {
    public static void main(String[] args) {
        try (BufferedReader br = new BufferedReader(new InputStreamReader(System.in))) {
            String[] strs = br.readLine().split(" ");
            char[] chars = strs[0].toCharArray();
            char target = strs[1].charAt(0);
            int[] result = new int[chars.length];
            int count = 1000;
            for (int i = 0; i < chars.length; i++) {
                if (target == chars[i]) {
                    count = 0;
                    result[i] = count;
                    count++;
                    continue;
                }
                result[i] = count;
                count++;
            }

            count = 1000;
            for (int j = chars.length - 1; j >= 0; j--) {
                if (target == chars[j]) {
                    count = 0;
                    result[j] = count;
                    count++;
                    continue;
                }
                if (result[j] > count) {
                    result[j] = count;
                }
                count++;
            }

            for (int i : result) {
                System.out.print(i + " ");
            }


        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/main/kotlin/sc02/Q10.kt
package jp.co.kopher.sc02

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader

fun main() {
    try {
        BufferedReader(InputStreamReader(System.`in`)).use { br ->
            val strs = br.readLine().split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            val chars = strs[0].toCharArray()
            val target = strs[1][0]
            val result = IntArray(chars.size)
            var count = 1000
            for (i in chars.indices) {
                if (target == chars[i]) {
                    count = 0
                    result[i] = count
                    count++
                    continue
                }
                result[i] = count
                count++
            }

            count = 1000
            for (j in chars.indices.reversed()) {
                if (target == chars[j]) {
                    count = 0
                    result[j] = count
                    count++
                    continue
                }
                if (result[j] > count) {
                    result[j] = count
                }
                count++
            }
            for (i in result) {
                print("$i ")
            }
        }
    } catch (e: IOException) {
        throw RuntimeException(e)
    }
}